### PR TITLE
(PCP-797) Add task-download-connect-timeout, task-download-timeout

### PR DESCRIPTION
**Pull request by MikaelSmith**
Configure task download timeouts based on new options `task-download-connect-timeout` and `task-download-timeout`.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
I'm looking into how to test this. I can probably setup something that doesn't respond to connection attempts or stops sending after the connection handshake, but not sure what will be easy to work into tests yet.

**Comment by MikaelSmith** [comment]
One option for testing was to add 1s delay to all traffic to the master node via
```
tc qdisc add dev eth0 root netem delay 1000ms
```
(reset via `tc qdisc del dev eth0 root netem`). Got
```
2018-04-23 16:27:25.238692 WARN  puppetlabs.pxp_agent.modules.task:301 - Downloading the task file from the master-uri 'https://puppet:8140' failed. Reason: File download server side error: Connection timed out after 1006 milliseconds
```
with both `task-download-connect-timeout: 1` and `task-download-timeout: 1`. That's probably not a good thing to put in tests though.

File: lib/src/modules/task.cc [hunk 1]
```diff
@@ -245,6 +249,11 @@ static std::string createUrlEndpoint(const lth_jc::JsonContainer& uri) {
     return url;
 }
 
+struct DownloadConfig {
```
Comment (nicklewis) [inline lib/src/modules/task.cc]: This struct isn't used.

**Comment by nicklewis** [comment]
Seems like this works in my testing.

**Pull request merged**